api/server.go

```diff
@@ -468,6 +474,62 @@ func (s *server) addressessAddressEventsHandler(jc jape.Context) {
 	jc.Encode(events)
 }
 
+func (s *server) addressessAddressEventsUnconfirmedHandler(jc jape.Context) {
+	var address types.Address
+	if jc.DecodeParam("address", &address) != nil {
+		return
+	}
+
+	events, err := s.e.AddressUnconfirmedEvents(address)
+	if jc.Check("failed to get unconfirmed address events", err) != nil {
+		return
+	}
+
+	jc.Encode(events)
+}
+
+func (s *server) eventsIDHandler(jc jape.Context) {
+	var id types.Hash256
+	if jc.DecodeParam("id", &id) != nil {
+		return
+	}
+
+	events, err := s.e.Events([]types.Hash256{id})
+	if err != nil {
+		return
+	} else if len(events) > 0 {
+		jc.Encode(events[0])
+		return
+	}
+
+	{
+		v1, v2 := s.cm.PoolTransactions(), s.cm.V2PoolTransactions()
+
+		relevantV1 := v1[:0]
```

Comment:
these loops are redundant, it looks like you wanted to filter at one point but removed it maybe?